### ISSUE TITLE
Gitpod lauch does not work: flask command not found

### ISSUE THREAD

**Issue by fnielsen** [WDscholia/scholia#1801]
**Describe the bug**
Gitpod lauch does not work: flask command not found

**To Reproduce**
1. Go to https://gitpod.io/#https://github.com/WDscholia/scholia
2. Error message: "bash: flask: command not found

**Expected behavior**

**Screenshots**
Flask should start in the Gitpod Window.

**Additional context**
Perhaps `python -m flask ...`?

**Comment by Daniel-Mietchen** [comment]
I am looking into this and can reproduce it.

Here is what I get:

```bash
 HISTFILE=/workspace/.gitpod/cmd-0 history -r; {
export FLASK_APP=runserver.py

} && {
python -m pip install -r requirements.txt 

} && {
flask run -h 0.0.0.0 -p 8100    

}
gitpod /workspace/scholia $  HISTFILE=/workspace/.gitpod/cmd-0 history -r; {
> export FLASK_APP=runserver.py
> 
> } && {
> python -m pip install -r requirements.txt 
> 
> } && {
> flask run -h 0.0.0.0 -p 8100    
> 
> }
Requirement already satisfied: python-dateutil in /home/gitpod/.pyenv/versions/3.8.12/lib/python3.8/site-packages (from -r requirements.txt (line 1)) (2.8.2)
Collecting docopt
  Downloading docopt-0.6.2.tar.gz (25 kB)
Collecting flask
  Downloading Flask-2.0.2-py3-none-any.whl (95 kB)
     |████████████████████████████████| 95 kB 5.1 MB/s 
Collecting Flask-Bootstrap
  Downloading Flask-Bootstrap-3.3.7.1.tar.gz (456 kB)
     |████████████████████████████████| 456 kB 96.3 MB/s 
Collecting flask-mwoauth>=0.1.14
  Downloading flask-mwoauth-0.4.82.tar.gz (7.7 kB)
Collecting lxml>=4.6.5
  Downloading lxml-4.7.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (6.9 MB)
     |████████████████████████████████| 6.9 MB 98.8 MB/s 
Requirement already satisfied: requests in /home/gitpod/.pyenv/versions/3.8.12/lib/python3.8/site-packages (from -r requirements.txt (line 7)) (2.26.0)
Collecting simplejson
  Downloading simplejson-3.17.6-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (139 kB)
     |████████████████████████████████| 139 kB 72.1 MB/s 
Collecting werkzeug>=0.9
  Downloading Werkzeug-2.0.2-py3-none-any.whl (288 kB)
     |████████████████████████████████| 288 kB 90.6 MB/s 
Requirement already satisfied: urllib3>=1.25.1 in /home/gitpod/.pyenv/versions/3.8.12/lib/python3.8/site-packages (from -r requirements.txt (line 10)) (1.26.6)
Collecting feedparser
  Downloading feedparser-6.0.8-py3-none-any.whl (81 kB)
     |████████████████████████████████| 81 kB 16.1 MB/s 
Requirement already satisfied: six>=1.5 in /home/gitpod/.pyenv/versions/3.8.12/lib/python3.8/site-packages (from python-dateutil->-r requirements.txt (line 1)) (1.16.0)
Requirement already satisfied: Jinja2>=3.0 in /home/gitpod/.pyenv/versions/3.8.12/lib/python3.8/site-packages (from flask->-r requirements.txt (line 3)) (3.0.1)
Collecting itsdangerous>=2.0
  Downloading itsdangerous-2.0.1-py3-none-any.whl (18 kB)
Collecting click>=7.1.2
  Downloading click-8.0.3-py3-none-any.whl (97 kB)
     |████████████████████████████████| 97 kB 12.0 MB/s 
Collecting dominate
  Downloading dominate-2.6.0-py2.py3-none-any.whl (29 kB)
Collecting visitor
  Downloading visitor-0.1.3.tar.gz (3.3 kB)
Collecting mwoauth
  Downloading mwoauth-0.3.7-py3-none-any.whl (13 kB)
Collecting requests-oauthlib
  Downloading requests_oauthlib-1.3.1-py2.py3-none-any.whl (23 kB)
Collecting future
  Downloading future-0.18.2.tar.gz (829 kB)
     |████████████████████████████████| 829 kB 70.6 MB/s 
Requirement already satisfied: certifi>=2017.4.17 in /home/gitpod/.pyenv/versions/3.8.12/lib/python3.8/site-packages (from requests->-r requirements.txt (line 7)) (2021.5.30)
Requirement already satisfied: charset-normalizer~=2.0.0 in /home/gitpod/.pyenv/versions/3.8.12/lib/python3.8/site-packages (from requests->-r requirements.txt (line 7)) (2.0.4)
Requirement already satisfied: idna<4,>=2.5 in /home/gitpod/.pyenv/versions/3.8.12/lib/python3.8/site-packages (from requests->-r requirements.txt (line 7)) (3.2)
Collecting sgmllib3k
  Downloading sgmllib3k-1.0.0.tar.gz (5.8 kB)
Requirement already satisfied: MarkupSafe>=2.0 in /home/gitpod/.pyenv/versions/3.8.12/lib/python3.8/site-packages (from Jinja2>=3.0->flask->-r requirements.txt (line 3)) (2.0.1)
Collecting oauthlib
  Downloading oauthlib-3.2.0-py3-none-any.whl (151 kB)
     |████████████████████████████████| 151 kB 92.3 MB/s 
Collecting PyJWT<2.0.0,>=1.0.1
  Downloading PyJWT-1.7.1-py2.py3-none-any.whl (18 kB)
Building wheels for collected packages: docopt, Flask-Bootstrap, flask-mwoauth, future, sgmllib3k, visitor
  Building wheel for docopt (setup.py) ... done
  Created wheel for docopt: filename=docopt-0.6.2-py2.py3-none-any.whl size=13724 sha256=139af97fc4436803e2451acadeb07d861c9af10821c9df408e629f2a6704a4e8
  Stored in directory: /home/gitpod/.cache/pip/wheels/56/ea/58/ead137b087d9e326852a851351d1debf4ada529b6ac0ec4e8c
  Building wheel for Flask-Bootstrap (setup.py) ... done
  Created wheel for Flask-Bootstrap: filename=Flask_Bootstrap-3.3.7.1-py3-none-any.whl size=460124 sha256=66429b2e17ef898620601a57edf393436d42089f8cbfb1a7499860ff75de77eb
  Stored in directory: /home/gitpod/.cache/pip/wheels/f2/a3/85/fe8b65a65a447c9906e3b7edb7d9e6c74dfa9c8425c3dd3007
  Building wheel for flask-mwoauth (setup.py) ... done
  Created wheel for flask-mwoauth: filename=flask_mwoauth-0.4.82-py3-none-any.whl size=5890 sha256=aceacb70757a1a2affc56a662f5674eabac764fb9b5312fe0cfd1711382191af
  Stored in directory: /home/gitpod/.cache/pip/wheels/d7/79/3d/196726fa688f8fb2381306d9fc00935e07748442a2834ff21d
  Building wheel for future (setup.py) ... done
  Created wheel for future: filename=future-0.18.2-py3-none-any.whl size=491070 sha256=d7654b02ca9197d5d2d4dd4a447b82ec6a65e0389a48595f373f6ab724f7627f
  Stored in directory: /home/gitpod/.cache/pip/wheels/8e/70/28/3d6ccd6e315f65f245da085482a2e1c7d14b90b30f239e2cf4
  Building wheel for sgmllib3k (setup.py) ... done
  Created wheel for sgmllib3k: filename=sgmllib3k-1.0.0-py3-none-any.whl size=6065 sha256=38a53dc5527770079bde10928f63de0f437c4dd1c3618517b5031761829108be
  Stored in directory: /home/gitpod/.cache/pip/wheels/83/63/2f/117884c3b19d46b64d3d61690333aa80c88dc14050e269c546
  Building wheel for visitor (setup.py) ... done
  Created wheel for visitor: filename=visitor-0.1.3-py3-none-any.whl size=3943 sha256=8d29a24ee094aca0c8b935a91412fc9432f39f3573bf57e7485fe9510c03ee48
  Stored in directory: /home/gitpod/.cache/pip/wheels/d3/40/52/5dae7760434a82caf8b8f88323029188b2d4ea3ac1235e550a
Successfully built docopt Flask-Bootstrap flask-mwoauth future sgmllib3k visitor
Installing collected packages: oauthlib, werkzeug, requests-oauthlib, PyJWT, itsdangerous, click, visitor, sgmllib3k, mwoauth, future, flask, dominate, simplejson, lxml, flask-mwoauth, Flask-Bootstrap, feedparser, docopt
Successfully installed Flask-Bootstrap-3.3.7.1 PyJWT-1.7.1 click-8.0.3 docopt-0.6.2 dominate-2.6.0 feedparser-6.0.8 flask-2.0.2 flask-mwoauth-0.4.82 future-0.18.2 itsdangerous-2.0.1 lxml-4.7.1 mwoauth-0.3.7 oauthlib-3.2.0 requests-oauthlib-1.3.1 sgmllib3k-1.0.0 simplejson-3.17.6 visitor-0.1.3 werkzeug-2.0.2
WARNING: You are using pip version 21.2.4; however, version 22.0.3 is available.
You should consider upgrading via the '/home/gitpod/.pyenv/versions/3.8.12/bin/python -m pip install --upgrade pip' command.
bash: flask: command not found
```

**Comment by Daniel-Mietchen** [comment]
So ```flask``` is clearly installed and its installation does not raise an error.

Here is what I get when pip installing it yet again:

```bash
pip install flask
Requirement already satisfied: flask in /home/gitpod/.pyenv/versions/3.8.12/lib/python3.8/site-packages (2.0.2)
Requirement already satisfied: Werkzeug>=2.0 in /home/gitpod/.pyenv/versions/3.8.12/lib/python3.8/site-packages (from flask) (2.0.2)
Requirement already satisfied: Jinja2>=3.0 in /home/gitpod/.pyenv/versions/3.8.12/lib/python3.8/site-packages (from flask) (3.0.1)
Requirement already satisfied: itsdangerous>=2.0 in /home/gitpod/.pyenv/versions/3.8.12/lib/python3.8/site-packages (from flask) (2.0.1)
Requirement already satisfied: click>=7.1.2 in /home/gitpod/.pyenv/versions/3.8.12/lib/python3.8/site-packages (from flask) (8.0.3)
Requirement already satisfied: MarkupSafe>=2.0 in /home/gitpod/.pyenv/versions/3.8.12/lib/python3.8/site-packages (from Jinja2>=3.0->flask) (2.0.1)
WARNING: You are using pip version 21.2.4; however, version 22.0.3 is available.
You should consider upgrading via the '/home/gitpod/.pyenv/versions/3.8.12/bin/python3 -m pip install --upgrade pip' command.
```

Afterwards, flask runs fine:

```bash
flask run -h 0.0.0.0 -p 8100    
 * Serving Flask app 'runserver.py' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
 * Running on http://10.0.2.100:8100/ (Press CTRL+C to quit)
192.168.56.74 - - [07/Feb/2022 17:30:36] "GET /?vscodeBrowserReqId=1644255036070 HTTP/1.1" 200 -
192.168.56.74 - - [07/Feb/2022 17:30:36] "GET /static/css/bootstrap.min.css HTTP/1.1" 200 -
192.168.56.74 - - [07/Feb/2022 17:30:36] "GET /static/css/scholia.css HTTP/1.1" 200 -
192.168.56.74 - - [07/Feb/2022 17:30:36] "GET /static/jquery.dataTables.min.css HTTP/1.1" 200 -
192.168.56.74 - - [07/Feb/2022 17:30:36] "GET /static/images/scholia_wordmark.svg HTTP/1.1" 200 -
192.168.56.74 - - [07/Feb/2022 17:30:36] "GET /static/jquery.min.js HTTP/1.1" 200 -
192.168.56.74 - - [07/Feb/2022 17:30:36] "GET /static/js/popper.min.js HTTP/1.1" 200 -
192.168.56.74 - - [07/Feb/2022 17:30:36] "GET /static/js/bootstrap.min.js HTTP/1.1" 200 -
192.168.56.74 - - [07/Feb/2022 17:30:36] "GET /static/d3.v5.min.js HTTP/1.1" 200 -
192.168.56.74 - - [07/Feb/2022 17:30:36] "GET /static/d3-scale-chromatic.v1.min.js HTTP/1.1" 200 -
192.168.56.74 - - [07/Feb/2022 17:30:36] "GET /static/bootstrap-autocomplete.min.js HTTP/1.1" 200 -
192.168.56.74 - - [07/Feb/2022 17:30:36] "GET /static/jquery.dataTables.min.js HTTP/1.1" 200 -
192.168.56.74 - - [07/Feb/2022 17:30:36] "GET /static/scholia.js HTTP/1.1" 200 -
```

**Comment by Daniel-Mietchen** [comment]
One reason I see for this is that maybe the ```init``` part of ```.gitpod.yml``` has not finished yet when the ```command``` part starts.

So here is our current ```gitpod.yml```:
```yml
image:
  file: .gitpod.Dockerfile

ports:
- port: 8100
  onOpen: open-preview
tasks:
- before: |
    export FLASK_APP=runserver.py
  init: >
    python -m pip install -r requirements.txt 
  command: >
    flask run -h 0.0.0.0 -p 8100    


```


And here is an overview of how to ensure that the ```init``` part completes before the ```command``` part kicks in: https://community.gitpod.io/t/gitpod-yml-and-making-sure-init-completes-before-command/2818/2 .

**Comment by fnielsen** [comment]
What about `python -m flask run 0.0.0.0 -p 8100`?

**Comment by Daniel-Mietchen** [comment]
```bash
gitpod /workspace/scholia $ python -m flask run 0.0.0.0 -p 8100
Usage: python -m flask run [OPTIONS]
Try 'python -m flask run --help' for help.

Error: Got unexpected extra argument (0.0.0.0)
```

**Comment by Daniel-Mietchen** [comment]
I tried a few more things in https://github.com/WDscholia/scholia/tree/gitpod-configuration-1801 , none of which solved the problem completely.
Interestingly, the version without ```.gitpod.yml``` worked fine, except that this requires manual opening of the preview.

I also found a Flask template for Gitpod configuration at https://github.com/gitpod-io/template-python-flask but have not tested it.
